Common/Common.Implementation/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// <copyright company="Tarcha & Ivchenko Company">
+//      Copyright (c) 2015, All Right Reserved
+// </copyright>
+// <author>Ivan Ivchenko</author>
+// <author>Myroslava Tarcha</author>
+
+using System;
+
+namespace Common.Implementation.Extensions
+{
+    /// <summary>
+    /// Provides a set of static methods to extend functionality of the <see cref="System.String"/>.
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Throws if a specified string is null, empty, or consists only of white-space characters.
+        /// </summary>
+        /// <param name="source">String under test.</param>
+        /// <param name="name">The name of the parameter that is under test.</param>
+        public static void ValidateEmpty(this string source, string name)
+        {
+            source.ValidateNull(name);
+
+            if (string.IsNullOrWhiteSpace(source))
```

Comment:
You're right sweetheart =)
I just want to divide exceptions: 
* first if the string is NULL -> ArgumentNullException
* second if the string is "" or "   " (let it be an 'empty string') -> ArgumentException

so I used DRY

Why do I did that?
I think that nulll and empty string could be caused by different use cases. Null means that program logic fail to get a real value and empty string means it is a real value but the value is not appropriate.